tests/test_main.rs

```diff
@@ -19,9 +19,60 @@ fn test_run_script() {
     Command::cargo_bin("pn")
         .unwrap()
         .current_dir(&temp_dir)
-        .arg("run")
-        .arg("test")
+        .args(["run", "test"])
         .assert()
         .success()
         .stdout("hello world\n");
 }
+
+#[test]
+fn test_workspace_root() {
+    let temp_dir = tempfile::tempdir().unwrap();
+    let temp_dir_path = temp_dir.path();
+    fs::write(
+        temp_dir_path.join("package.json"),
+        r#"{"scripts": {"test": "echo hello from workspace root"}}"#,
+    )
+    .unwrap();
+    fs::write(
+        temp_dir_path.join("pnpm-workspace.yaml"),
+        r#"packages: ["packages/*"]"#,
+    )
+    .unwrap();
+    let project_foo_path = temp_dir_path.join("packages/foo");
+    fs::create_dir_all(&project_foo_path).unwrap();
+    fs::write(
+        project_foo_path.join("package.json"),
+        r#"{"scripts": {"test": "echo hello from foo"}}"#,
+    )
+    .unwrap();
+
+    Command::cargo_bin("pn")
+        .unwrap()
+        .current_dir(project_foo_path)
+        .args(["--workspace-root", "run", "test"])
+        .assert()
+        .success()
+        .stdout("hello from workspace root\n");
+}
+
+#[test]
+fn test_workspace_root_not_found() {
+    let temp_dir = tempfile::tempdir().unwrap();
+    let package_json_path = temp_dir.path().join("package.json");
+    fs::write(
+        package_json_path,
+        r#"{"scripts": {"test": "echo hello world"}}"#,
+    )
+    .unwrap();
+
+    let assertion = Command::cargo_bin("pn")
+        .unwrap()
+        .current_dir(&temp_dir)
+        .args(["--workspace-root", "run", "test"])
+        .assert()
+        .failure();
+    let output = assertion.get_output();
+    let stderr = String::from_utf8(output.stderr.clone()).unwrap();
+    assert!(stderr.contains("--workspace-root may only be used in a workspace"));
```

Comment:
You would want to log the stderr before doing assertion because you would want to know its value in case of test failure:

```suggestion
    eprintln!("STDERR:\n{stderr}\n");
    assert!(stderr.contains("--workspace-root may only be used in a workspace"));
```

Don't worry about it tainting `cargo test` messages, because you would see the `eprintln` log if and only if the test fails.